index.js

```diff
@@ -1,95 +1,119 @@
-const { getConfig } = require('./lib/config')
-const { isTriggerableBranch } = require('./lib/triggerable-branch')
-const {
-  findReleases,
-  generateReleaseInfo,
-  createRelease,
-  updateRelease
-} = require('./lib/releases')
-const { findCommitsWithAssociatedPullRequests } = require('./lib/commits')
-const { sortPullRequests } = require('./lib/sort-pull-requests')
-const log = require('./lib/log')
-const core = require('@actions/core')
-
-module.exports = app => {
-  app.on('push', async context => {
-    const config = await getConfig({
-      app,
-      context,
-      configName: core.getInput('config-name')
-    })
-
-    if (config === null) return
-
-    // GitHub Actions merge payloads slightly differ, in that their ref points
-    // to the PR branch instead of refs/heads/master
-    const ref = process.env['GITHUB_REF'] || context.payload.ref
-
-    const branch = ref.replace(/^refs\/heads\//, '')
-
-    if (!isTriggerableBranch({ branch, app, context, config })) {
-      return
-    }
-
-    const { draftRelease, lastRelease } = await findReleases({ app, context })
-    const {
-      commits,
-      pullRequests: mergedPullRequests
-    } = await findCommitsWithAssociatedPullRequests({
-      app,
-      context,
-      branch,
-      lastRelease
-    })
-
-    const sortedMergedPullRequests = sortPullRequests(
-      mergedPullRequests,
-      config['sort-by'],
-      config['sort-direction']
-    )
-
-    const releaseInfo = generateReleaseInfo({
-      commits,
-      config,
-      lastRelease,
-      mergedPullRequests: sortedMergedPullRequests,
-      version: core.getInput('version') || undefined,
-      tag: core.getInput('tag') || undefined,
-      name: core.getInput('name') || undefined
-    })
-
-    const shouldDraft = core.getInput('publish').toLowerCase() !== 'true'
-
-    let createOrUpdateReleaseResponse
-    if (!draftRelease) {
-      log({ app, context, message: 'Creating new release' })
-      createOrUpdateReleaseResponse = await createRelease({
-        context,
-        releaseInfo,
-        shouldDraft,
-        config
-      })
-    } else {
-      log({ app, context, message: 'Updating existing release' })
-      createOrUpdateReleaseResponse = await updateRelease({
-        context,
-        draftRelease,
-        releaseInfo,
-        shouldDraft,
-        config
-      })
-    }
-
-    setActionOutput(createOrUpdateReleaseResponse)
-  })
-}
-
-function setActionOutput(releaseResponse) {
-  const {
-    data: { id: releaseId, html_url: htmlUrl, upload_url: uploadUrl }
-  } = releaseResponse
-  if (releaseId && Number.isInteger(releaseId))
-    core.setOutput('id', releaseId.toString())
-  if (htmlUrl) core.setOutput('html_url', htmlUrl)
-  if (uploadUrl) core.setOutput('upload_url', uploadUrl)
-}
+const { getConfig } = require('./lib/config')
```

Comment:
Looks like a line ending conflict

Reply:
Feel free to submit a PR that adds a gitattributes that prevents this conflict 😓

Reply:
Oh, sorry, I have merged this to my master, to test with my another PR.. I wasn't aware of this type conflict, how can I resolve this?

Reply:
https://help.github.com/en/github/using-git/configuring-git-to-handle-line-endings

Reply:
I've reversed my merge, it was wrong adding that feature with this one... the conflict doesn't exist anymore...

Reply:
Something's wrong with index.js, even when normalized to LF it keeps saying (on VSCode at least) that it's with CRLF

Reply:
My git is configured to be LF but the file is commited as CRLF by someone else (only index.js), i've downloaded the zip file to test this, on VScode this is the line break for this file:

![line-break](https://user-images.githubusercontent.com/32045525/78565869-da838780-77f4-11ea-9c24-0d0cbc782685.png)

Even when I run `git add --renormalize .` it doesn't change at all. 
Do you know what it can be? 

PS: i've read the docs and didn't find anything referring to something like this.

I can change the file line break to LF manually, but this creates a "broken" diff.

Reply:
created #476 this is why an `gitattributes` file is important to avoid these issues.

Reply:
Sorry for the hassle, can I ask you to clean up the commits by perhaps squashing and rebase with master?

Reply:
Looks like you manged to "duplicate" the change set.

I'd suggest rebase and squash the commits as we use merge pull request on this repo, your commits seems rather noisy as well 😅

Reply:
Basically reset to 891d2cd and force push should do the trick :)

Reply:
Done, I've used pull before push :fearful: